src/main/java/org/prebid/server/bidder/BidderMediaTypeProcessor.java

```diff
@@ -0,0 +1,131 @@
+package org.prebid.server.bidder;
+
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.collections4.SetUtils;
+import org.prebid.server.auction.model.AuctionContext;
+import org.prebid.server.spring.config.bidder.model.MediaType;
+import org.prebid.server.util.ObjectUtil;
+
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class BidderMediaTypeProcessor {
+
+    private final BidderCatalog bidderCatalog;
+
+    public BidderMediaTypeProcessor(BidderCatalog bidderCatalog) {
+        this.bidderCatalog = Objects.requireNonNull(bidderCatalog);
+    }
+
+    public BidRequest process(BidRequest bidRequest,
```

Comment:
What about returning Result<BidRequest>? Then we don't need AuctionContextParameter, don't need to return null.